### PR TITLE
[flake8-pyi] Skip Annotated and Literal for `string-or-bytes-too-long (PYI053)`

### LINKED ISSUE #12995: `string-or-bytes-too-long (PYI053)` flags and removes long strings in `Literal`, producing invalid type
In the example reproduction below, 

* List of keywords you searched for before creating this issue. Write them down here so that others can find this issue more easily and help provide feedback.
  string-or-bytes-too-long PYI053
  
* A minimal code snippet that reproduces the bug.
```py
def foo(bar: typing.Literal["a", "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"]):...
```
becomes
```py
import typing

def foo(bar: typing.Literal["a", ...]):...
```

Not only is this an invalid type form (https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportInvalidTypeForm) but even if autofixed to `Literal["a"]` that would fundamentally change the parameter's type.

I assume this is an oversight as it's meant to replace long default values ~~and is not flagged by flake8-pyi~~ .

* The command you invoked (e.g., `ruff /path/to/file.py --fix`), ideally including the `--isolated` flag.
ruff check --isolated --select=PYI053

* The current Ruff settings (any relevant sections from your `pyproject.toml`).
not relevant

* The current Ruff version (`ruff --version`).
ruff 0.6.1

### PR DESCRIPTION
**Pull request by dylwil3**
PYI053 now skips strings which appear inside `Literal[...]` or `Annotated[...]`.

This is a modification of #13002, which skipped any type annotation for this rule. So, on the one hand, this PR allows the check to proceed in some cases that the prior PR skipped (e.g. return type annotations wrapped - unnecessarily, for a stub file - in quotes), and, on the other hand, this PR skips certain strings that the prior PR did not (e.g. type alias assignments to a `Literal` or `Annotated` type.)

In order to implement this change, the PR adds a new semantic flag for PEP593 annotations (i.e. `typing.Annotated[...]`) 

Closes #12995 (hopefully!)

### CODE REVIEW

File: crates/ruff_python_semantic/src/model.rs [hunk 1]
```diff
@@ -2229,6 +2235,14 @@ bitflags! {
         /// [PEP 257]: https://peps.python.org/pep-0257/#what-is-a-docstring
         const ATTRIBUTE_DOCSTRING = 1 << 26;
 
+        /// The model is in a `typing.Annotated` annotation.
+        ///
+        /// For example, the model could be visiting `int` or "A" in
+        /// ```python
+        /// def f(x: Annotated[int, "A"]): ...
+        /// ```
+        const TYPING_ANNOTATED_PEP593 = 1<<27;
```
Comment (AlexWaygood) [inline crates/ruff_python_semantic/src/model.rs:2244]: oof... but I have been trying to think of a better name for 10 minutes and I've got nothing 😆

**Comment by AlexWaygood** [comment]
I know this has been waiting on me for a little bit -- have to call it for the day now, but will try to land it tomorrow!

**Comment by dylwil3** [comment]
> I know this has been waiting on me for a little bit -- have to call it for the day now, but will try to land it tomorrow!

No worries at all!

File: crates/ruff_python_semantic/src/model.rs (hunk 1 above)
Comment (dhruvmanila) [reply crates/ruff_python_semantic/src/model.rs:2244]: Seems like a reasonable name given other variants although I leave this up to you to decide, maybe you come up with a better name in the morning ;)

Comment (dhruvmanila) [reply crates/ruff_python_semantic/src/model.rs:2244]: Aside, it would be useful to provide a link to PEP 593 in the documentation above @dylwil3

**Comment by dylwil3** [comment]
@AlexWaygood if it was possible to meekly whisper a nudge in the form of a notification, I would, but github allows for only one volume. Let me know if there's anything else you'd like me to do on this one 😄

**Comment by charliermarsh** [comment]
@dylwil3 - Just a heads up (in case it wasn't mentioned elsewhere) that Alex is on vacation this week so may be a few more days before he gets to it. (It's totally fine that you nudged and in fact your nudge was very gracious!)

**Comment by dylwil3** [comment]
Closing in favor of #14797

**Pull request closed without merging**